Libplanet/Store/IStore.cs

```diff
@@ -38,9 +38,14 @@ public interface IStore
         /// </summary>
         /// <param name="namespace">The namespace of the index that contains block hashes to
         /// iterate.</param>
+        /// <param name="offset">The starting point to return block hashes.</param>
+        /// <param name="limit">The maximum number of block hashes to get.</param>
         /// <returns>Block hashes in the index of the <paramref name="namespace"/>, in ascending
         /// order; the genesis block goes first, and the tip block goes last.</returns>
-        IEnumerable<HashDigest<SHA256>> IterateIndex(string @namespace);
+        IEnumerable<HashDigest<SHA256>> IterateIndex(
+            string @namespace,
+            int offset = 0,
+            int limit = int.MaxValue);
```

Comment:
Defaulting it to `int.MaxValue` seems not that robust to me.  How about making it required, or changing its type to `int?` and defaulting it to `null`?

Reply:
Since the type is `int`, changing the default doesn't seem to make much difference. Or should it be a `long` type?

Reply:
My concern was that the current way is difficult to distinguish the intention of “unlimiting” from “limiting up to a kind of very large number, which is the maximum possible value of 32-bit integer.”  Also I thought it's a C-ish trick to me.

In the current way, if we're willing to make a store backend on SQLite, we need to determine if it's intended to be unlimited, by comparing it to `int.MaxValue`, in order to avoid SQL like `OFFSET 0 LIMIT 2147483647` and simply put `OFFSET 0`, which differ each other.

My suggestion is simple: how about `int? limit = null`?

Reply:
Ok, got it. Fixed!